File: client/src/types/marketplace.ts
export interface MarketplaceListing {
    id: string;
    agent_id: string;
    user_id: string;
    title: string;
    summary: string;
    category: string;
    price_cents: number;
    status: 'draft' | 'active' | 'archived';
    featured: boolean;
    cover_image_url?: string;
    download_count: number;
    created_at: string;
    updated_at: string;
    agent_name?: string;
    agent_description?: string;
    agent_icon_name?: string;
    agent_skill_ids?: string[];
    publisher_name?: string;
    publisher_avatar_url?: string;
    downloaded: boolean;
}

export interface CreateListingPayload {
    agent_id: string;
    title: string;
    summary?: string;
    category?: string;
    price_cents?: number;
    status?: 'draft' | 'active' | 'archived';
}

export interface UpdateListingPayload {
    title?: string;
    summary?: string;
    category?: string;
    price_cents?: number;
    status?: 'draft' | 'active' | 'archived';
}

export interface MarketplaceDownloadResponse {
    already_downloaded: boolean;
    agent?: any; // We can use the Agent type here if imported
}

File: server/models/marketplace.go
package models

import "time"

type MarketplaceListing struct {
	ID             string    `json:"id"`
	AgentID        string    `json:"agent_id"`
	UserID         string    `json:"user_id"`
	Title          string    `json:"title"`
	Summary        string    `json:"summary"`
	Category       string    `json:"category"`
	PriceCents     int       `json:"price_cents"`
	Status         string    `json:"status"`
	Featured       bool      `json:"featured"`
	CoverImageURL  string    `json:"cover_image_url,omitempty"`
	DownloadCount  int       `json:"download_count"`
	CreatedAt      time.Time `json:"created_at"`
	UpdatedAt      time.Time `json:"updated_at"`
	// Joined for API response (optional)
	AgentName      string   `json:"agent_name,omitempty"`
	AgentDesc      string   `json:"agent_description,omitempty"`
	AgentIconName  string   `json:"agent_icon_name,omitempty"`
	AgentSkillIDs  []string `json:"agent_skill_ids,omitempty"`
	PublisherName  string   `json:"publisher_name,omitempty"`
	PublisherAvatar string  `json:"publisher_avatar_url,omitempty"`
	Downloaded     bool     `json:"downloaded"`
}

type MarketplaceDownload struct {
	ID        string    `json:"id"`
	ListingID string    `json:"listing_id"`
	UserID    string    `json:"user_id"`
	CreatedAt time.Time `json:"created_at"`
}
